SDWebImage/SDImageCacheConfig.m

```diff
@@ -50,7 +50,7 @@ - (id)copyWithZone:(NSZone *)zone {
     config.maxCacheSize = self.maxCacheSize;
     config.maxMemoryCost = self.maxMemoryCost;
     config.maxMemoryCount = self.maxMemoryCount;
-    config.namespacePrefix = [self.namespacePrefix copyWithZone:zone];
```

Comment:
immutable NSString seems always return `self` when call `copyWithZone:`. So acutally this does not cause any problem.

Write like this is just by practice, since sometimes refactor may loose something and introduce a mutable `NSMutableString`...So I guess change this or not seems not really important :)

Reply:
🤔 Emm, `namespacePrefix`'s attribute is `copy`, so it can handle `NSString` and `NSMutableString` correctly. What's your meaning about `introduce a mutable NSMutableString`? you mean somebody may subclass `SDImageCacheConfig`? or change the `namespacePrefix` property from `NSString` to `NSMutableString`? If it is, people should change the attribute of `copy` to `strong` and override its own `copyWithZone`, and then use `mutableCopyWithZone` I think. What do you think? 😂

Reply:
For example, someone write like this `_namespacePrefix = another.namespacePrefix`, this will cause potential issue on mutable string.

I write like this, just follow the old style of implementation `copyWithZone:`, which should better pass the `zone` params to every object property and call their coorespond attributed. For copy property, call `copyWithZone:`, for strong/weak property, call `xxx` getter method..`NSString` here use copy attribute, so I just use `copyWithZone:`, regardless of what actual class it is.

But actually, since now that zone params has no effect, you can directlly call `self.namespacePrefix = another.namespacePrefix` as well. Change or not seems no effect :)

Reply:
Emm, I think the potential issue is that set the `_namespacePrefix = [NSMutableString ....]` because of set the ivar directly, other words, in `copyWithZone` of `SDImageCacheConfig.m`, `self.namespacePrefix = [NSMutableString ..]` or `self.namespacePrefix = another.namespacePrefix` always works, which means `self.namespacePrefix` is always `NSString`.
And a small performance if we do `config.namespacePrefix = [self.namespacePrefix copyWithZone:zone]` is we call `copy` two times, because `namespacePrefix`'s attribute is `copy`, the code can translate into:

```
config->_namespacePrefix = [ [self.namespacePrefix copyWithZone:zone] copy];
```

But, yep, it's ok if you think we can ignore these trivial effects.
I can revert these changes this evening. 😂